core/src/main/java/org/kohsuke/stapler/StaplerResponseWrapper.java

```diff
@@ -0,0 +1,336 @@
+package org.kohsuke.stapler;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.io.PrintWriter;
+import java.io.Writer;
+import java.net.URL;
+import java.util.Locale;
+
+import javax.servlet.ServletException;
+import javax.servlet.ServletOutputStream;
+import javax.servlet.http.Cookie;
+import javax.servlet.http.HttpServletRequest;
+
+import org.kohsuke.stapler.StaplerRequest;
+import org.kohsuke.stapler.StaplerResponse;
+import org.kohsuke.stapler.export.Flavor;
+
+import net.sf.json.JsonConfig;
+
+/**
+ * A basic wrapper for a StaplerResponse, e.g. in order to override some method.
+ * This simply delegates all method calls to the wrapped instance.
+ * 
+ * @since TODO
+ */
+@SuppressWarnings("deprecation")
+public abstract class StaplerResponseWrapper implements StaplerResponse {
+    private final StaplerResponse wrapped;
+    
+    public StaplerResponseWrapper(StaplerResponse wrapped) {
```

Comment:
You'd still need to forward all the calls to some `StaplerResponse` somewhere. Let's say you wanted to `@Inject StaplerResponse`, you couldn't set the current response and you'd have to provide a wrapper, for example, to look it up from a `ThreadLocal` variable every time.